corehq/apps/settings/views.py

```diff
@@ -538,17 +541,11 @@ def dispatch(self, request, *args, **kwargs):
 class TwoFactorResetView(TwoFactorSetupView):
     urlname = 'reset'
 
-    form_list = (
-        ('welcome_reset', HQEmptyForm),
-        ('method', HQTwoFactorMethodForm),
-        ('generator', HQTOTPDeviceForm),
-        ('sms', HQPhoneNumberForm),
-        ('call', HQPhoneNumberForm),
-        ('validation', HQDeviceValidationForm),
-    )
-
     def get(self, request, *args, **kwargs):
         default_device(request.user).delete()
+        # django-two-factor-auth caches the default device on the user so clear that after deleting
+        from two_factor.utils import USER_DEFAULT_DEVICE_ATTR_NAME
+        delattr(request.user, USER_DEFAULT_DEVICE_ATTR_NAME)
```

Comment:
Should we have a test verifying this behavior?

Reply:
Ended up understanding this a bit better, and decided to avoid the `default_device` method two_factor provides since it has a side effect of caching the device. Instead I now manually fetch the device and delete it and added tests for this behavior in fd5332d7b42a46a7f5f99363944789176400fb63 (the commit messsage also has some context).